tests/replication/_base.py

```diff
@@ -69,7 +69,7 @@ def prepare(self, reactor, clock, hs):
         self.worker_hs.get_datastore().db_pool = hs.get_datastore().db_pool
 
         self.test_handler = self._build_replication_data_handler()
-        self.worker_hs.replication_data_handler = self.test_handler
+        self.worker_hs._replication_data_handler = self.test_handler
```

Comment:
Can `replication_data_handler` be passed into `setup_test_homeserver` (similar to the changes made bleow in `tests/rest/client/v1/test_presence.py`)?

Reply:
This cannot be easily done because `_build_replication_data_handler` instantiates `TestReplicationDataHandler` with the homeserver, so you end up with a chicken/egg problem.